OpenMage/magento-lts
# In product list all swatches are always shown even if they are not available when "Display Out of Stock Products" is disabled

### Description (*)

In the product list, all swatches are always shown even if they are not available.
You can verify this on the PDP because only the available ones are displayed correctly.

Tested with both enabled and disabled stock inventory configuration "Display Out of Stock Products"
![inventory_settings](https://github.com/user-attachments/assets/7647e266-1fde-4a49-9d8b-7946d4efc877)

When "Display Out of Stock Products" is enabled shown correctly like this:
![show_out_of_stock](https://github.com/user-attachments/assets/d2848333-968d-49d9-baeb-4395f640436d)

When "Display Out of Stock Products" is disabled all swatches are always shown even if they are not available.
With this patch shown correctly like this:

![not_out_of_stock](https://github.com/user-attachments/assets/08e93a55-4c2d-4394-a0f6-b4827530ede5)

The bug occurs when the " Display Out of Stock Products" configuration is disabled.

### Related Pull Requests


### Fixed Issues (if relevant)


### Manual testing scenarios (*)

1. Create configurable product with options as swatch such as color
2. Set stock unavailable for one color option

Example with sample data:
1. In product configurable Name: "Nolita cami" SKU "wbk000c" set all children products with color pink unavailable.

In the product list, all colors are always shown even if they are not available.
You can verify this on the PDP because only the available ones are displayed correctly.

![pdp](https://github.com/user-attachments/assets/6b33df90-c68f-48c5-977d-fb321497e248)
![pl](https://github.com/user-attachments/assets/50f6779f-ad65-436d-bcfb-126fd57c2feb)



### Questions or comments


### Contribution checklist (*)
